lib/job/arRepositoryCsvExportJob.class.php

```diff
@@ -73,16 +73,17 @@ protected function exportResults($path)
   {
     $itemsExported = 0;
 
-    $resultSet = QubitSearch::getInstance()->index->getType('QubitRepository')->search($this->search->getQuery(false, false));
+    $search = QubitSearch::getInstance()->index->getType('QubitRepository')->createSearch($this->search->getQuery(false, false));
 
     $writer = new csvRepositoryExport($path, null, 10000);
     $writer->loadResourceSpecificConfiguration('QubitRepository');
 
-    foreach ($resultSet as $hit)
+    // Scroll through results, caching IDs in file, and iterate through IDs in file (given scroll result will expire)
+    foreach ($identifierIteratorFile = arElasticSearchPluginUtil::cacheScrolledSearchResultIdentifiersAsIterableFileObject($search) as $id)
     {
-      if (null === $resource = QubitRepository::getById($hit->getId()))
+      if (null === $resource = QubitRepository::getById($id))
       {
-        $this->error($this->i18n->__('Cannot fetch repository, id: '.$hit->getId()));
+        $this->error($this->i18n->__('Cannot fetch repository, id: '. $id));
```

Comment:
I don't know if we go with spaces around the `.` operator or not, but try not to use both options at the same time ;)

Reply:
Haha I think I picked that up with Drupal standards or something (no space next to string, but space between var and concatenator). Maybe I'll go with no space as I'm not sure we have a standard.

Reply:
(Actually I'll use sprintf as that's cleaner anyways.)